File: CS/Messenger/SendingAndReceivingCustomMessagesUserControl.cs
using DevExpress.Mvvm;
using DevExpress.Utils.MVVM;
using DevExpress.XtraEditors;
using System;
using System.Windows.Forms;

namespace DxSampleMessenger {
    public partial class SendingAndReceivingCustomMessagesUserControl : XtraUserControl {
        public SendingAndReceivingCustomMessagesUserControl() {
            InitializeComponent();
            #region SetUp
            MVVMContext mvvmContext = new MVVMContext();
            mvvmContext.ContainerControl = this;
        
            SimpleButton sendMessageButton = new SimpleButton();
            sendMessageButton.Text = "Send Message";
            sendMessageButton.Dock = DockStyle.Top;
            sendMessageButton.Parent = this;
            #endregion SetUp

            #region #sendingAndReceivingCustomMessages
            // add another view
            CustomMessageAwareView msgView = new CustomMessageAwareView();
            msgView.Parent = sendMessageButton.Parent;
            msgView.BringToFront();
            // start listening the ViewModel's custom messages  in View
            msgView.RegisterAsCustomMessageRecepient();
        
            mvvmContext.ViewModelType = typeof(ViewModelWithCustomMessage);
            // UI bindings for SendCustomMessage commands
            var fluentAPI = mvvmContext.OfType<ViewModelWithCustomMessage>();
            fluentAPI.BindCommand(sendMessageButton, x => x.SendCustomMessage());
            #endregion #sendingAndReceivingCustomMessages
        }
        #region CleanUp
        void OnDisposing() {
            var context = MVVMContext.FromControl(this);
            if(context != null) context.Dispose();
        }
        #endregion CleanUp
        public class ViewModelWithCustomMessage {
            Random rnd = new Random();
            public void SendCustomMessage() {
                Messenger.Default.Send(new CustomMessage() { Parameter = rnd.Next(0, 100) });
            }
            // Custom Message related to this ViewModel
            public class CustomMessage {
                public object Parameter { get; set; }
                public override string ToString() {
                    return string.Format("CustomMessage: {0}!", Parameter);
                }
            }
        }
        public class CustomMessageAwareView : XtraUserControl {
            MemoEdit memo;
            public CustomMessageAwareView() {
                this.Padding = new Padding(0, 0, 0, 4);
                this.Dock = DockStyle.Fill;
                memo = new MemoEdit();
                memo.Dock = DockStyle.Fill;
                memo.Parent = this;
            }
            public void RegisterAsCustomMessageRecepient() {
                Messenger.Default.Register<ViewModelWithCustomMessage.CustomMessage>(this, OnCustomMessage);
            }
            void OnCustomMessage(ViewModelWithCustomMessage.CustomMessage message) {
                memo.Text += (message.ToString() + Environment.NewLine);
                memo.SelectionStart = memo.Text.Length;
                memo.ScrollToCaret();
            }
        }
    }
}

File: VB/Messenger/SendingAndReceivingCustomMessagesUserControl.vb
Imports DevExpress.Mvvm
Imports DevExpress.Utils.MVVM
Imports DevExpress.XtraEditors
Imports System
Imports System.Windows.Forms

Namespace DxSampleMessenger
    Partial Public Class SendingAndReceivingCustomMessagesUserControl
        Inherits DevExpress.XtraEditors.XtraUserControl
        Public Sub New() 
            InitializeComponent()
            Dim mvvmContext As New MVVMContext()
            mvvmContext.ContainerControl = Me
        
            Dim sendMessageButton As New SimpleButton()
            sendMessageButton.Text = "Send Message"
            sendMessageButton.Dock = DockStyle.Top
            sendMessageButton.Parent = Me

            ' add another view
            Dim msgView As New CustomMessageAwareView()
            msgView.Parent = sendMessageButton.Parent
            msgView.BringToFront()
            ' start listening the ViewModel's custom messages  in View
            msgView.RegisterAsCustomMessageRecepient()
        
            mvvmContext.ViewModelType = GetType(ViewModelWithCustomMessage)
            ' UI bindings for SendCustomMessage commands
            Dim fluentAPI = mvvmContext.OfType(Of ViewModelWithCustomMessage)()
            fluentAPI.BindCommand(sendMessageButton, Sub(x) x.SendCustomMessage())
        End Sub
         Private Sub OnDisposing()
            Dim context = MVVMContext.FromControl(Me)
            If context IsNot Nothing Then
                context.Dispose()
            End If
        End Sub
        Public Class ViewModelWithCustomMessage
            Private rnd As New Random()
            Public Sub SendCustomMessage()
                Messenger.Default.Send(New CustomMessage() With {.Parameter = rnd.Next(0, 100)})
            End Sub
            ' Custom Message related to this ViewModel
            Public Class CustomMessage
                Public Property Parameter() As Object
                Public Overrides Function ToString() As String
                    Return String.Format("CustomMessage: {0}!", Parameter)
                End Function
            End Class
        End Class
        Public Class CustomMessageAwareView
            Inherits XtraUserControl
            Private memo As MemoEdit
            Public Sub New()
                Me.Padding = New Padding(0, 0, 0, 4)
                Me.Dock = DockStyle.Fill
                memo = New MemoEdit()
                memo.Dock = DockStyle.Fill
                memo.Parent = Me
            End Sub
            Public Sub RegisterAsCustomMessageRecepient()
                Messenger.Default.Register(Of ViewModelWithCustomMessage.CustomMessage)(Me, AddressOf OnCustomMessage)
            End Sub
            Private Sub OnCustomMessage(ByVal message As ViewModelWithCustomMessage.CustomMessage)
                memo.Text += (message.ToString() & Environment.NewLine)
                memo.SelectionStart = memo.Text.Length
                memo.ScrollToCaret()
            End Sub
        End Class
    	End Class
End Namespace